pages/1_orders_and_revenue.py

```diff
@@ -18,49 +18,163 @@
+        fig = px.line(orders_over_time, x='month', y='header_id')
+        fig.update_yaxes(title_text='Orders')
+        fig.update_xaxes(title_text='Month')
+        st.plotly_chart(fig, use_container_width=True)
 
     with row2_col1:
         st.markdown("**Percent of Successful Payments Over Time**")
+        successful_payments = data[data['header_status'] == 'completed'].groupby('month')['payment_id'].nunique()
+        total_payments = data.groupby('month')['payment_id'].nunique()
+        percent_successful = (successful_payments / total_payments * 100).fillna(0).reset_index()
+        fig = px.line(percent_successful, x='month', y='payment_id')
+        fig.update_yaxes(range=[0, 100], title_text='Percentage (%)')
+        fig.update_xaxes(title_text='Month')
+        st.plotly_chart(fig, use_container_width=True)
 
     with row2_col2:
         st.markdown("**New Customers Over Time**")
+        data['customer_created_month'] = data['customer_created_at'].dt.to_period('M').dt.to_timestamp()
+        new_customers_over_time = data.groupby('customer_created_month')['customer_id'].nunique().reset_index()
+        fig = px.line(new_customers_over_time, x='customer_created_month', y='customer_id')
+        fig.update_yaxes(title_text='Customers')
+        fig.update_xaxes(title_text='Month')
+        st.plotly_chart(fig, use_container_width=True)
+
+# Customer Table
+
+# Calculate Total Spend
+total_spend = data.groupby('customer_id')['total_amount'].sum().reset_index()
+
+# Calculate Total Orders
+total_orders = data.groupby('customer_id')['header_id'].nunique().reset_index()
+total_orders.rename(columns={'header_id': 'total_orders'}, inplace=True)
+
+# Calculate Total Refunds
+total_refunds = data.groupby('customer_id')['refund_amount'].sum().reset_index()
+
+# Calculate Last Order Date
+last_order_date = data.groupby('customer_id')['created_at'].max().reset_index()
+
+# Calculate Created Date
+created_date = data.groupby('customer_id')['customer_created_at'].min().reset_index()
+
+# Merge all the calculated fields
+customer_table = data[['customer_id', 'customer_name', 'customer_email', 'customer_city', 'customer_country']].drop_duplicates().reset_index(drop=True)
+customer_table = customer_table.merge(total_spend, on='customer_id', how='left')
+customer_table = customer_table.merge(total_orders, on='customer_id', how='left')
+customer_table = customer_table.merge(total_refunds, on='customer_id', how='left')
+customer_table = customer_table.merge(last_order_date, on='customer_id', how='left')
+customer_table = customer_table.merge(created_date, on='customer_id', how='left')
+
+# Rename columns
+customer_table.rename(columns={
+    'customer_id': 'ID',
+    'customer_name': 'Name',
+    'customer_email': 'Email',
+    'customer_city': 'City',
+    'customer_country': 'Country',
+    'total_amount': 'Total Spend',
+    'refund_amount': 'Total Refunds',
+    'created_at': 'Last Order Date',
+    'customer_created_at': 'Created Date',
```

Comment:
This is great, I think it would be interesting to also include total discounts. Would you be able to add that in as well?

Reply:
added!